src/app/chat/hooks/useStatusSubscription.ts

```
@@ -77,103 +94,136 @@ export function useStatusSubscription(userIds: string[]) {
 
+                if (!enabledRef.current) {
+                    return;
+                }
+
                 setStatuses(statusMap);
             }
         } catch (error) {
             logger.error("Failed to fetch statuses", toError(error));
         } finally {
+            if (!enabledRef.current) {
+                return;
```

Biome (2.4.14)

[error] Unsafe usage of 'return'.

(lint/correctness/noUnsafeFinally)